### ISSUE TITLE
How to configure Blitz to only purge reverse proxy caches, and not cache anything local to the webserver

### ISSUE THREAD

**Issue by tomdavies** [putyourlightson/craft-blitz#502]
Hi Ben 👋 , apologies if I've missed something or misunderstood the documentation, but I could really use an example for configuring Blitz to track cache tags _and_ element references, and handle invalidations in a reverse proxy cache (CloudFlare in our case), **but not actually cache any responses itself** 

i.e. I don't want to use either of the `YiiCacheStorage` or `FileStorage` drivers, since if a request misses the downstream edge cache, I _always_ want a fresh version generated from Craft, and not a locally cached copy returned.

I've tried every combination of config settings I can think of (I'm configuring everything via the config file) but I can't figure out a way to do this out of the box. At the moment, I've achieved this by using a "dummy" custom cache storage driver that does a no-op for every method it's required to implement, but that seems like a hack for something that feels like it should be a reasonably common use-case. (I guess I could set a very short `cacheDuration` but that also feels like a hack).

Relevant bits of my current `config/blitz.php`:

```php
<?php
use craft\helpers\App;

return [
    '*' => [
        'cachingEnabled' => true,
        'refreshMode' => 1,
        // The storage type to use.
        'cacheStorageType' => 'modules\frontend\blitz\drivers\storage\DummyStorage',

        // The purger type to use.
        'cachePurgerType' => 'putyourlightson\blitz\drivers\purgers\CloudflarePurger',

        // The purger settings (zone ID keys are site UIDs).
        'cachePurgerSettings' => [
            'zoneIds' => [
                // UID of the default site
                '<primary-site-uid>' => [
                    'zoneId' => App::env('CLOUDFLARE_CACHE_ZONE_ID'),
                ],
            ],
            'authenticationMethod' => 'apiToken',
            'apiToken' => App::env('CLOUDFLARE_CACHE_PURGE_TOKEN'),
        ],
    ],
];

```

FWIW I'm running
Craft 4.4.8
Blitz 4.4.4

**Comment by bencroker** [comment]
Using the `DummyStorage` class sounds ideal, is it working for you as expected? The storage drivers are intentionally extensible, so this doesn’t feel like a hack to me. In theory, we could make this class part of core, but you’re the first person to ever ask about this feature.

**Comment by tomdavies** [comment]
> you’re the first person to ever ask about this feature

Fair enough! 

Yes, my dummy driver works fine, just wanted to make sure I wasn’t missing something

**Comment by tomdavies** [comment]
For context about why I was trying to do this, my intention here was that:

1. I’m in a load balanced env in prod (so file caching doesn’t make a ton of sense), with a pretty large site (1000s of pages) and with a CloudFlare business plan in front of it (so I’ve got lots of options for configuration on CF, and it’s not super likely that CF will start randomly dropping my caches as it seems to on cheaper plans) 
2. I’ve found stacking multiple full page caches (edge + web server) hard to debug when I’ve hit issues in the past

**Comment by bencroker** [comment]
> Yes, my dummy driver works fine, just wanted to make sure I wasn’t missing something

I think that’s the perfect solution.

1. If it was me, I’d still tend to cache pages in a Redis store or something similar to prevent the origin server having to (re)process requests each time there’s a cache MISS from a Cloudflare edge node. I suppose that depends somewhat on how geographically distributed your site visitors are and how you’ve configured CF.
2. I can understand that. I find that leaving the Blitz comments in place, which also means disabling HTML minification in Cloudflare, can help when trying to figure out where the request was handled.

**Comment by bencroker** [comment]
FYI I’ve added a feature branch with a `DummyStorage` class.

You can test this by running composer require "putyourlightson/craft-blitz:dev-feature/dummy-storage-driver as 4.7.0".

**Comment by bencroker** [comment]
Released in [4.7.0](https://github.com/putyourlightson/craft-blitz/releases/tag/4.7.0).
